### PR TITLE
BUG: Fix for pandas.to_datetime reports incorrect index when failing

### LINKED ISSUE #59298: BUG: pandas.to_datetime reports incorrect index when failing.
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
pandas as pd
valid = '2024-01-01'
invalid = 'N'
S = pd.Series([valid]*45_000 + [invalid])
T = pd.to_datetime(S, format='%Y-%m-%d', exact=True) 

################################
# Traceback redacted.
################################
ValueError: time data "N" doesn't match format "%Y-%m-%d", at position 1. You might want to try:
    - passing `format` if your strings have a consistent format;
    - passing `format='ISO8601'` if your strings are all ISO8601 but not necessarily in exactly the same format;
    - passing `format='mixed'`, and the format will be inferred for each element individually. You might want to use `dayfirst` alongside this.
```

### Issue Description

As you can see a ValueError is raised. That is fine. 

The description says it happens for item at postition 1, however, which is not correct. It happens for item at position 45000.

### Expected Behavior

The correct position would be reported to the user. I'd like to do S[<position>] and be able to see why it raised valueError.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit                : d9cdd2ee5a58015ef6f4d15c7226110c9aab8140
python                : 3.11.8.final.0
python-bits           : 64
OS                    : Linux
OS-release            : 6.9.7-amd64
Version               : #1 SMP PREEMPT_DYNAMIC Debian 6.9.7-1 (2024-06-27)
machine               : x86_64
processor             : 
byteorder             : little
LC_ALL                : None
LANG                  : C.UTF-8
LOCALE                : en_US.UTF-8

pandas                : 2.2.2
numpy                 : 2.0.1
pytz                  : 2024.1
dateutil              : 2.9.0.post0
setuptools            : 68.1.2
pip                   : 24.1.1
Cython                : None
pytest                : None
hypothesis            : None
sphinx                : None
blosc                 : None
feather               : None
xlsxwriter            : None
lxml.etree            : None
html5lib              : None
pymysql               : None
psycopg2              : None
jinja2                : None
IPython               : None
pandas_datareader     : None
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : None
bottleneck            : None
dataframe-api-compat  : None
fastparquet           : None
fsspec                : None
gcsfs                 : None
matplotlib            : None
numba                 : None
numexpr               : None
odfpy                 : None
openpyxl              : None
pandas_gbq            : None
pyarrow               : None
pyreadstat            : None
python-calamine       : None
pyxlsb                : None
s3fs                  : None
scipy                 : None
sqlalchemy            : None
tables                : None
tabulate              : None
xarray                : None
xlrd                  : None
zstandard             : None
tzdata                : 2024.1
qtpy                  : None
pyqt5                 : None

</details>

### PR DESCRIPTION
**Pull request by maushumee**
- [x] closes #59298 
- [x] Tests added and passed
- [x] All code checks passed
- [x] Added type annotations
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by maushumee** [comment]
As per the discussion on the linked issue, removed the index from the error message.

File: pandas/tests/tools/test_to_datetime.py [hunk 1]
```diff
@@ -3500,6 +3482,15 @@ def test_unknown_tz_raises():
         to_datetime([dtstr])
 
 
+def test_unformatted_input_raises():
+    valid, invalid = "2024-01-01", "N"
+    S = Series([valid] * 45000 + [invalid])
```
Comment (mroeschke) [inline pandas/tests/tools/test_to_datetime.py:3487]:
```suggestion
    ser = Series([valid] * 2 + [invalid])
```

File: pandas/tests/tools/test_to_datetime.py [hunk 2]
```diff
@@ -3500,6 +3482,15 @@ def test_unknown_tz_raises():
...
+    msg = 'time data "N" doesn\'t match format "%Y-%m-%d"'
+
+    with pytest.raises(ValueError, match=msg):
+        to_datetime(S, format="%Y-%m-%d", exact=True)
```
Comment (mroeschke) [inline pandas/tests/tools/test_to_datetime.py:3491]:
```suggestion
        to_datetime(ser, format="%Y-%m-%d", exact=True, cache=True)
```

File: pandas/tests/tools/test_to_datetime.py (hunk 1 above)
Comment (maushumee) [reply pandas/tests/tools/test_to_datetime.py:3487]: Agreed. Missed the start_caching_at var in the code. Thanks for this suggestion!

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @maushumee

**Pull request merged**